### ISSUE TITLE
Copy / move file to different version?

### ISSUE THREAD

**Issue by hagenw** [audeering/audbackend#212]
We have `copy_file()` and `move_file()` for `audbackend.interface.Versioned` and `audbackend.interface.Maven`. But in both cases it is only possible to copy/move a file to the same version it was originally stored with.
One can still achieve the desired result by first downloading it and then uploading it again, but the idea behind `move_file()` and `copy_file()` was to avoid this.

**Comment by frankenjoe** [comment]
Yes that makes sense. I guess the simplest solution would you like to drop the option to copy / move all versions of a file and introduce `version_src` as a mandatory argument and have `version_dst` as an optional argument.

**Comment by hagenw** [comment]
I also like the option to copy/move all versions of a file at once.
I would propose the following:
* stay with the name `version` for specifying the source version
* add `version_dst` to specify a destination version
* if `version` is not given copy/move all files. If `version_dst` is given in this case, raise a `ValueError`

This would have also the advantage that we can stick with the current solution for now and add `version_dst` only if we need it at a later stage without breaking any API.

**Comment by frankenjoe** [comment]
Yes, sounds good to me.
